test/test_cuda.py

```diff
@@ -1458,38 +1456,40 @@ def test_multinomial_ext(self):
         samples = probs.multinomial(1000000, replacement=True)
         self.assertGreater(probs[samples].min().item(), 0)
 
-    @staticmethod
-    def mute():
-        os.dup2(os.open(os.devnull, os.O_WRONLY), sys.stderr.fileno())
-
-    def _spawn_method(self, method, arg):
-        ctx = mp.get_context("spawn")
-        with ctx.Pool(1, initializer=self.mute) as pool:
-            errors = pool.map(method, [arg])
-            for e in errors:
-                if 'device-side assert triggered' not in str(e):
-                    self.fail(e)
-
-    @staticmethod
-    def _test_multinomial_invalid_probs_cuda(probs):
+    def _spawn_test_multinomial_invalid_probs_cuda(self, probs):
+        import subprocess
         try:
-            with torch.random.fork_rng(devices=[0]):
-                torch.multinomial(probs.to('cuda'), 2, replacement=True)
-                torch.cuda.synchronize()
-            return False  # Should not be reached
-        except RuntimeError as e:
-            return e
+            p = subprocess.Popen([sys.executable, '-c', f"""\
+import sys
+import torch
+from torch._six import inf, nan
+try:
+    with torch.random.fork_rng(devices=[0]):
+        torch.multinomial(torch.tensor({probs}).to('cuda'), 2, replacement=True)
+        torch.cuda.synchronize()
+    sys.exit(-1) # Should not be reached
+except RuntimeError as e:
+    sys.exit(-2)
+"""], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+            out, err = p.communicate(timeout=10)
+        except subprocess.TimeoutExpired as e:
+            p.kill()
+            out, err = p.communicate()
+        expected_messages = [
+            'device-side assert triggered',  # CUDA
+            'Assertion',  # CUDA
+            'HSA_STATUS_ERROR_EXCEPTION',  # ROCm
+            'Device-side assertion'  # ROCm
+        ]
+        self.assertTrue(any([msg in out or msg in err for msg in expected_messages]))
 
-    @slowTest
```

Comment:
Please keep slow test decorator in place

Reply:
Is it really a slow test?  @ngimel showed it as `Ran 1 test in 24.941s`.  Generally speaking, what is the rationale for marking a test as slow?

Reply:
Restored slowTest decorator in 25986b6841dfd15e67fa3013082cad4023ae50cc.

Reply:
> Is it really a slow test?

Average test runtime for TestCUDA class is 0.4 sec, which makes 24s test slow relative to other tests in the same class.

> Generally speaking, what is the rationale for marking a test as slow?

PyTorch testsuite consists of 40+k tests with an average test run time of .4 sec, which makes it possible to finish testing in a bit over 4 hours. If average time per testcase were 20 sec, it would have taken 9 days to finish test suite.  Annotating tests as slow gives better control of test runtime, but at the same time allows infrequent runs of slow test suite.